prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -323,6 +324,48 @@ class FhirTransformerTests {
         assertThat(newValue[0].primitiveValue()).isEqualTo("abc123")
     }
 
+    @Test
+    fun `test transform with value set`() {
+        val bundle = Bundle()
+        bundle.id = "bundle1"
+        val resource = Patient()
+        resource.id = "abc123"
+        bundle.addEntry().resource = resource
+        val resource2 = Patient()
+        resource2.id = "def456"
+        bundle.addEntry().resource = resource2
+        val resource3 = Patient()
+        resource3.id = "jkl369"
+        bundle.addEntry().resource = resource3
+
+        val patientElement = FhirTransformSchemaElement(
+            "patientElement",
+            value = listOf("%resource.id"),
+            resource = "%resource",
+            bundleProperty = "%resource.id",
+            valueSet = sortedMapOf(
+                Pair("abc123", "ghi789"),
+                Pair("def456", "")
```

Comment:
What happens if there is a pair with `null`. Is this an impossible state? If so, where do we test/verify that?

Reply:
I think this is just compiler enforced since `valueSet` is of type `SortedMap<String, String>`

Reply:
Correct, this is immediately flagged as a type mismatch in the IDE if you attempt to define either the key or the value as anything other than `String`.